anoma/namada-interface
# feat/835 -  Tx details in extension

Resolves #835 

- [x] Add ability to deserialize Tx in `shared` and `sdk` - we have a `to_bytes()` which will return borsh-serialized values that can be deserialized into typed props
- [x] Add query in extension to access pending Tx, and retrieve these details by deserializing Tx bytes
- [x] Display details in `ApproveSignTx`
- [x] Display JSON details when clicking `View Data`
- [x] Add query in extension to access pending `Sign Arbitrary` data, which will return a string that we can attempt to parse as JSON, display in `ApproveSignArbitrary` `View Data` view
- [x] Update SDK docs

### TESTING

__NOTE__ Best to start with a clean install of the extension and new account to test the entire flow. Also, this should probably be tested with the Luminare testnet!

#### Testing Tx details
- Install new copy of extension based on this branch
- Create/Import account (if you want to see RevealPk being added to batch, create an account here, then transfer funds via the faucet at https://faucet.luminara.icu/)
- Once connected in Namadillo, create some Bonds, then submit for approval (if this is the first time this account is used on chain, you will see a RevealPk tx in the list - NOTE that we don't have to display this, I'm just iterating through all Tx in a batch)
- Click `View Data` to see the entire object of the batch Tx, go back & approve
- Ensure that submitting transaction works!

#### Testing Sign-Arbitrary details
- In Namadillo, enable `Sign Arbitrary` in settings, and submit a sign arbitrary request by clicking the icon
- In `View Data`, you should see the string that was passed to the extension for signing
- You can also enter valid JSON into text area, then when you submit it for signing, `View Data` should parse and format this

### TODO

Styles will be addressed in a follow-up PR
